app/js/app.js

```diff
@@ -133,4 +133,14 @@ alertaApp.config(['config', '$authProvider',
       clientId: config.client_id,
       authorizationEndpoint: config.keycloak_url+'/auth/realms/'+config.keycloak_realm+'/protocol/openid-connect/auth'
     });
+    $authProvider.oauth2({
+      name: 'pingfederate',
+      url: config.endpoint+'/auth/pingfederate',
+      redirectUri: window.location.origin+'/',
+      clientId: config.client_id,
+      authorizationEndpoint: config.pingfederate_url,
+      pfidpadapterid: 'kerberos',
```

Comment:
Is `pfidpadapterid` needed? It's not a supported OAuth2 option so I wonder if it is relevant.

Reply:
It's needed in pingfederate to avoid the login form from sso. In grafana I can configure it like 
pingfederate_url+?pfidpadapterid=kerberos
However here I cannot do that because in flask app the other attribute are appended with `?` and I end up with two `?` in url which is invalid. Hence I had to include this here.

Reply:
I just tested this and the `pfidpadapterid=kerberos` is not being sent in the request so it can't possibly be doing what you think it's doing...
```
GET /?response_type=code&client_id=INSERT-CLIENT-ID-HERE&redirect_uri=http://localhost:8000/ HTTP/1.1
Host: localhost:4500
Connection: keep-alive
Cache-Control: max-age=0
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.181 Safari/537.36
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9
Cookie: __utma=1.491622643.1477996808.1510696298.1510703097.58; __utmz=1.1501360019.43.1.utmcsr=(direct)|utmccn=(direct)|utmcmd=(none); _ga=GA1.1.491622643.1477996808; _gid=GA1.1.1613387615.1522356786
```
The only supported options for OAuth2 are listed here ... https://github.com/sahat/satellizer/blob/master/src/oauth2.ts#L6

Reply:
Hi. Thanks for the response. Here is the documentation for `pfidpadapterid=kerberos` https://documentation.pingidentity.com/pingfederate/pf84/#concept_authorizationEndpoint.html#concept_authorizationEndpoint

This option affects the way provider shows the login screen. using kerberos enables direct login and user is not presented with any form.

Reply:
You are correct. Looks like it's not getting passed. I just checked with my provider. First time it presents users with a option to choose kerberos or login form. I had selected kerberos hence never saw that form again. 

Can you help me in passing this parameter also? I mean it's not a must but a good to have as it will avoid that one time extra screen.

Reply:
I debugged it. Looks like my auth endpoint is going as `auth_url?response_type=code&client_id=maas_sso&redirect_uri=my_app_url/`
They want it auth url to be `auth_url?pfidpadapterid=kerberos` then followed by other option.

Reply:
Looks like you just need to add `requiredUrlParams: ['pfidpadapterid', 'scope'],` to authProvider config object.

That is, it should look like this ...
```
    $authProvider.oauth2({
      name: 'pingfederate',
      url: config.endpoint+'/auth/pingfederate',
      redirectUri: window.location.origin+'/',
      clientId: config.client_id,
      authorizationEndpoint: config.pingfederate_url,
      requiredUrlParams: ['pfidpadapterid', 'scope'],
      scope: 'openid+profile+email',
      pfidpadapterid: 'kerberos'
    });
```
This config will give you a URL ...
`...&redirect_uri=http://localhost:8000/&pfidpadapterid=kerberos&scope=openid+profile+email`

Is that what you needed?

Reply:
this is exactly what I wanted. I will commit these changes. thanks a lot. I wasn't aware of this.

Reply:
Neither was I until I looked at the source code. This feature does not seem to be documented very well.